clients/cli/src/main/java/org/apache/gravitino/cli/ReadTableCSV.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.cli;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.gravitino.rel.Column;
+
+public class ReadTableCSV {
+
+  private enum ExpectedColumns {
+    NAME("Name"),
+    DATATYPE("Datatype"),
+    COMMENT("Comment"),
+    NULLABLE("Nullable"),
+    AUTOINCREMENT("AutoIncrement"),
+    DEFAULTVALUE("DefaultValue"),
+    DEFAULTTYPE("DefaultType");
+
+    private final String name;
+
+    ExpectedColumns(String name) {
+      this.name = name;
+    }
+
+    public String getName() {
+      return name;
+    }
+  }
+
+  public Column[] columns(Map<String, List<String>> tableData) {
+    List<String> names = tableData.get(ExpectedColumns.NAME.getName());
+    List<String> datatypes = tableData.get(ExpectedColumns.DATATYPE.getName());
+    List<String> comments = tableData.get(ExpectedColumns.COMMENT.getName());
+    List<String> nullables = tableData.get(ExpectedColumns.NULLABLE.getName());
+    List<String> autos = tableData.get(ExpectedColumns.AUTOINCREMENT.getName());
+    List<String> defaultTypes = tableData.get(ExpectedColumns.DEFAULTTYPE.getName());
+    List<String> defaulValues = tableData.get(ExpectedColumns.DEFAULTVALUE.getName());
+    int size = names.size();
+    Column[] columns = new Column[size];
+
+    for (int i = 0; i < size; i++) {
+      String columnName = names.get(i);
+      String datatype = datatypes.get(i);
+      String columnComment = comments.get(i);
+      boolean nullable = nullables.get(i).equals("true");
+      boolean auto = autos.get(i).equals("true");
+      String defaultValue = defaulValues.get(i);
+      String defaultType = defaultTypes.get(i);
+
+      if (defaultType == null || defaultType.isEmpty()) {
+        defaultType = datatype;
+      }
```

Comment:
It is the correct logic. If the default datatype is not supplied then use the data type of the column for the default value's datatype.

Reply:
I could be wrong though. My first impression was that if "data type" is not provided, we are gonna use the "default data type" as the "data type".
The current logic says otherwise, i.e. if "default data type" is not provided, we use the provided "data type" as the default.
That was confusing to me.
Thanks for clarification.

Reply:
For columns with default values, most of the time, the data type of the default value will be the same as the data type of the column itself, so there's no real need to specify it.